src/SortableList.js

```diff
@@ -87,8 +87,7 @@ export default class SortableList extends Component {
   componentWillReceiveProps(nextProps) {
     const {data, order} = this.state;
     let {data: nextData, order: nextOrder} = nextProps;
-
-    if (data && nextData && !shallowEqual(data, nextData)) {
+    if (data && nextData && Object.keys(data).toString() !== Object.keys(nextData).toString()) {
```

Comment:
In case when values of data changes we need layout recalculation, for example
`oldData = {1: 'Some text. Some text. Some text. Some text. Some text.'}`
`new Data = {1: 'Some text. Some text. Some text. Some text. Some text. Some text. Some text. Some text. Some text. Some text.'}`

here a key was not changed, but value changed and a new row will be higher...

Reply:
And generally about this commit, does is it really bottle neck for performance in your apps? I suppose no, because unlikely data is  frequently changing

Reply:
Is there a way to recalculate the layout _without_ changing the keys? It's not the performance but the screen flash with the rows remounting... any change (typing, etc) causes it to get the new keys.

Reply:
I think it maybe possible to implement it better, this was not in my mind when I wrote this component, because this feature was not need in our app, I added it after someone in issues asked me.

Currently, the component uses same rows for getting layout and showing to user, it firstly renders rows with right: 9999 and then when new layouts calculated shows them to user. For eliminating flash it should separately render rows for layout calculation and for user presentation